### PR TITLE
Reword the README's intro

### PR DESCRIPTION
**Pull request by scotts**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,15 +2,21 @@
...
-and fed to ML models. TorchCodec needs a working FFMPEG installation (that the
-user provides) and uses it as a library to do the decoding. We use FFMPEG
-because of the diverse video formats it supports as well as its popularity.
-
-Under the hood, TorchCodec uses Pytorch's C++ custom ops to wrap C++ code that
-does the memory management of packets, frames and codecs. TorchCodec hides
-that complexity under its API (see example usage below).
+TorchCodec is a Python library for decoding videos into PyTorch tensors. It aims 
+to be fast, easy to use and well integrated into the PyTorch ecosystem. If you 
+want to use PyTorch to train ML models on videos, TorchCodec how you turn those
```
Comment (ahmadsharif1) [inline README.md:7]: > TorchCodec how you turn those

grammar seems off

File: README.md [hunk 2]
```diff
@@ -2,15 +2,21 @@
...
-decode video frames to PyTorch Tensors. These tensors can then be transformed
-and fed to ML models. TorchCodec needs a working FFMPEG installation (that the
-user provides) and uses it as a library to do the decoding. We use FFMPEG
-because of the diverse video formats it supports as well as its popularity.
-
-Under the hood, TorchCodec uses Pytorch's C++ custom ops to wrap C++ code that
-does the memory management of packets, frames and codecs. TorchCodec hides
-that complexity under its API (see example usage below).
+TorchCodec is a Python library for decoding videos into PyTorch tensors. It aims 
+to be fast, easy to use and well integrated into the PyTorch ecosystem. If you
```
Comment (ahmadsharif1) [inline README.md:6]: hyphenated seems better

Comment (scotts) [reply README.md:6]: Which phrase?

File: README.md [hunk 3]
```diff
@@ -2,15 +2,21 @@
...
+We achieve these capabilities through:
+
+* Pythonic APIs that mirror Python and PyTorch conventions.
+* Relying on [FFMPEG](https://www.ffmpeg.org/) to do the decoding. TorchCodec 
+  uses the version of FFMPEG you already have installed. FMPEG is a mature 
+  library with broad coverage available on most systems. It is, however, not 
+  easy to use. TorchCodec abstracts FFMPEG's complexity to ensure it used 
+  correctly and efficiently.
+* Implementing the core logic as 
+  [PyTorch custom operators](https://pytorch.org/tutorials/advanced/custom_ops_landing_page.html).
```
Comment (NicolasHug) [inline README.md:19]: I think we should keep that out from this intro. This is an implementation details that users do not need to care about. Most users do not know what a custom op is. We don't want them to think they have to understand what custom ops are, or understand the linked tutorial, to use torchcodec.

File: README.md [hunk 4]
```diff
@@ -2,15 +2,21 @@
...
-that complexity under its API (see example usage below).
+TorchCodec is a Python library for decoding videos into PyTorch tensors. It aims 
+to be fast, easy to use and well integrated into the PyTorch ecosystem. If you 
+want to use PyTorch to train ML models on videos, TorchCodec how you turn those 
+videos into data.
+
+We achieve these capabilities through:
+
+* Pythonic APIs that mirror Python and PyTorch conventions.
+* Relying on [FFMPEG](https://www.ffmpeg.org/) to do the decoding. TorchCodec
```
Comment (NicolasHug) [inline README.md:13]: We should try to spell it `FFmpeg`, as the authors recommend Should be `FFmpeg` (they care about it https://ffmpeg.org//legal.html)

(below in other places as well).

File: README.md (hunk 2 above)
Comment (NicolasHug) [reply README.md:6]: oxford comma after "use", I think

**Review by NicolasHug — APPROVED** [review]
Some comments / suggestions below. I'll approve to unblock, but I would insist on removing mentions to custom ops in that intro.

Additionally, it might be worth  mirroring these changes into https://github.com/pytorch/torchcodec/blob/main/docs/source/index.rst

File: README.md [hunk 5]
```diff
@@ -2,15 +2,21 @@
...
+want to use PyTorch to train ML models on videos, TorchCodec how you turn those 
+videos into data.
+
+We achieve these capabilities through:
+
+* Pythonic APIs that mirror Python and PyTorch conventions.
+* Relying on [FFMPEG](https://www.ffmpeg.org/) to do the decoding. TorchCodec 
+  uses the version of FFMPEG you already have installed. FMPEG is a mature 
+  library with broad coverage available on most systems. It is, however, not 
+  easy to use. TorchCodec abstracts FFMPEG's complexity to ensure it used
```
Comment (NicolasHug) [inline README.md:16]: I think it's missing a verb: "is used" or "gets used"

File: README.md (hunk 3 above)
Comment (scotts) [reply README.md:19]: That's fair. When we have tested `torch.compile()` support, I think we'll want to mention that.

File: README.md (hunk 2 above)
Comment (scotts) [reply README.md:6]: Oh,  @ahmadsharif1 is referring to "easy to use." That should only be hyphenated when it modifieds a noun, as it is a phrase that acts as an adjective. But we're using the phrase in its verb form, so it shouldn't be hyphenated.

And I grudgingly accept that I am in the minority of disliking Oxford commas. :)

File: README.md [hunk 6]
```diff
@@ -2,15 +2,21 @@
...
+TorchCodec is a Python library for decoding videos into PyTorch tensors. It aims
+to be fast, easy to use, and well integrated into the PyTorch ecosystem. If you
+want to use PyTorch to train ML models on videos, TorchCodec is how you turn
+those videos into data.
+
+We achieve these capabilities through:
+
+* Pythonic APIs that mirror Python and PyTorch conventions.
+* Relying on [FFmpeg](https://www.ffmpeg.org/) to do the decoding. TorchCodec
+  uses the version of FFmpeg you already have installed. FMPEG is a mature
```
Comment (NicolasHug) [inline README.md:14]:
```suggestion
  uses the version of FFmpeg you already have installed. FFmpeg is a mature
```

File: README.md (hunk 2 above)
Comment (ahmadsharif1) [reply README.md]: Yes, was talking about "easy-to-use".

Otherwise it could be parsed as "fast, easy" as well, leaving the "to use" a bit confusing.

**Review by ahmadsharif1 — APPROVED** [review]

**Pull request merged**